src/librustc_typeck/collect.rs

```diff
@@ -1149,6 +1168,95 @@ fn type_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     }
 }
 
+fn find_existential_constraints<'a, 'tcx>(
```

Comment:
comment needed -- it'd be great to see some examples of what this code is doing :)

Reply:
ah, I see, is it checking that all uses are compatible?